// esbuild-plugin-copy/dist/lib/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PLUGIN_EXECUTED_FLAG = exports.formatAssets = exports.verboseLog = exports.ensureArray = void 0;
const chalk_1 = __importDefault(require("chalk"));
function ensureArray(item) {
    return Array.isArray(item) ? item : [item];
}
exports.ensureArray = ensureArray;
function verboseLog(msg, verbose, lineBefore = false) {
    if (!verbose) {
        return;
    }
    console.log(chalk_1.default.blue(lineBefore ? '\ni' : 'i'), msg);
}
exports.verboseLog = verboseLog;
function formatAssets(assets) {
    return ensureArray(assets)
        .filter((asset) => asset.from && asset.to)
        .map(({ from, to, keepStructure = true }) => ({
        from: ensureArray(from),
        to: ensureArray(to),
        keepStructure,
    }));
}
exports.formatAssets = formatAssets;
exports.PLUGIN_EXECUTED_FLAG = 'esbuild_copy_executed';
